gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/BackendsMetricStats.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.clustermonitor;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.airlift.log.Logger;
+import io.trino.gateway.ha.config.MonitorConfiguration;
+import io.trino.gateway.ha.config.ProxyBackendConfiguration;
+import io.trino.gateway.ha.router.GatewayBackendManager;
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
+import org.weakref.jmx.MBeanExporter;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Singleton
+public class BackendsMetricStats
+{
+    private static final Logger log = Logger.get(BackendsMetricStats.class);
+    public static final int DEFAULT_METRIC_REFRESH_SECONDS = 30;
+
+    private final MBeanExporter exporter;
+    private final GatewayBackendManager gatewayBackendManager;
+    private final MonitorConfiguration monitorConfiguration;
+    private Map<String, BackendClusterMetricStats> statsMap = new ConcurrentHashMap<>();
+    private final ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+
+    @Inject
+    public BackendsMetricStats(GatewayBackendManager gatewayBackendManager, MBeanExporter exporter, MonitorConfiguration monitorConfiguration)
+    {
+        this.gatewayBackendManager = gatewayBackendManager;
+        this.exporter = exporter;
+        this.monitorConfiguration = monitorConfiguration;
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        int refreshSeconds = monitorConfiguration.getMetricRefreshSeconds();
+        log.info("Running periodic metric refresh with interval of %d seconds", refreshSeconds);
+        scheduledExecutor.scheduleAtFixedRate(() -> {
+            try {
+                initMetrics();
+            }
+            catch (Exception e) {
+                log.error(e, "Error refreshing backend metrics");
+            }
+        }, 0, refreshSeconds, TimeUnit.SECONDS);
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        scheduledExecutor.shutdownNow();
+    }
+
+    public void initMetrics()
```

Comment:
`initMetrics()` -> I think `init` is not quite right, maybe "refresh" or "update" ?